### PR TITLE
add mocha compilers option to transpile es2015

### PR DESCRIPTION
**Pull request by ryof**
I have a same issue as mentioned in #5 .
I think this issue will be fixed if this plugin passes compiler-option to mocha :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,6 +98,9 @@ class mochaPlugin {
               path: {
                 usage: 'Path for the tests for running tests in other than default "test" folder',
               },
+              compilers: {
+                usege: 'Compiler to use on Mocha',
```
Comment (laardee) [inline index.js]: typo, change usege to usage

**Comment by ryof** [comment]
@laardee I verified its functionality by running following command in my serverless project:
`sls invoke test --compilers js:babel-register`

You have to install these dependencies if you try that.
+ babel-register
+ babel-preset-es2015

and add this to package.json
```
"babel": {
  "presets": [
    "es2015"
  ]
}
```

Please point my mistakes out above if any, cause I'm just a beginner in any of node.js, babel or serverless...

**Comment by laardee** [comment]
@ryof would the project be something like this https://github.com/SC5/serverless-mocha-plugin/tree/babel-tests/test/test-service-node6.10-babel? I made that integration test for testing this feature. You can rebase it to your fork, it is in the 'babel-tests' branch. After rebase you can run only that test with `mocha test/integration6.10-babel.js -t 10000`

The test contains test and function templates that have some ES2015 features used. It first creates a test, then a function with test and then runs the tests.

File: index.js [hunk 2]
```diff
@@ -176,6 +179,22 @@ class mochaPlugin {
               mocha.reporter(reporter, reporterOptions);
             }
 
+            const compilers = myModule.options['compilers'];
+            if (compilers !== undefined) {
```
Comment (laardee) [inline index.js]: `if (typeof compilers !== 'undefined') {`

File: index.js [hunk 3]
```diff
@@ -176,6 +179,22 @@ class mochaPlugin {
...
+              const extensions = ['js'];
+              myModule.options['compilers'].split(' ').filter(function(e){return e!==''}).forEach(function (c) {
+                const idx = c.indexOf(':');
```
Comment (laardee) [inline index.js]: I would rather split the c with the first occurrence of `:` and use those parts, perhaps with reduce. 
```
c.split(/:(.+)/)
  .reduce(
```

File: index.js [hunk 4]
```diff
@@ -176,6 +179,22 @@ class mochaPlugin {
...
+                const ext = c.slice(0, idx);
+                const mod = c.slice(idx + 1);
+
+                if (mod[0] === '.') {
+                  mod = join(process.cwd(), mod);
```
Comment (laardee) [inline index.js]: Should the `join(process.cwd(), mod);` be `path.join(process.cwd(), mod);`?

File: index.js (hunk 3 above)
Comment (ryof) [inline index.js]: Split magic! I didn't know regexp can be used in split.
Please tell me the reason for using reduce here? 🙂

File: index.js (hunk 2 above)
Comment (ryof) [inline index.js]: fixed.

File: index.js (hunk 4 above)
Comment (ryof) [inline index.js]: fixed.

**Comment by ryof** [comment]
@laardee could you please review my changes again? :)

**Comment by laardee** [comment]
Oh, definitely, I try to review the changes today. Sorry about that.

**Comment by cdelgadob** [comment]
Is it possible to write the tests in TypeScript? Thanks

**Comment by hotgazpacho** [comment]
Is there a way to configure a default value for the `--compilers` flag? I'd really prefer to just run `sls invoke test` and not `sls invoke test --compilers js:babel-core/register`

**Pull request merged**